src/UiPath.Workflow.Runtime/Tracking/TrackingProvider.cs

```diff
@@ -324,6 +324,12 @@ private bool PostTrackingRecord(TrackingParticipant participant, RuntimeTracking
                 if (preparedRecord != null)
                 {
                     IAsyncResult result = participant.BeginTrack(preparedRecord, _timeoutHelper.RemainingTime(), PrepareAsyncCompletion(trackingCompleteCallback), this);
+
+                    if (preparedRecord is ActivityScheduledRecord record)
```

Comment:
this affects existing workflows and functionalities

Reply:
It should do nothing as nothing changes on the regular path. But you are right, we may call this method only when needed.